### ISSUE TITLE
Help wanted on constraints

### ISSUE THREAD

**Issue by Geli25** [donmccurdy/aframe-physics-system#110]
I would like to start by saying that I am very very new to aframe and its physics engine, so I am having trouble wrapping my head around this. This may be a very amateur question but I was hoping to get some help on this.

I am trying to make a rope-like constraint where the constraint is like a spring that bounces back, how can I do something like that? Do I need to apply the force?

**Comment by donmccurdy** [comment]
Hi @Geli25 —do you mean something like this example, using Cannon.JS?

http://schteppe.github.io/cannon.js/demos/constraints.html

^then open the **Links** or **Sphere chain** demos.

CannonJS is the physics engine I use in this project, and it's mostly designed for "rigid-body" physics, with solid objects. Things like ropes are harder, but not impossible. You need to start by chaining together a few (5-10) links, so they start to behave like a rope.

Plain JS code in Cannon.js for many point-to-point constraints: https://github.com/schteppe/cannon.js/blob/master/demos/constraints.html#L77-L111

HTML for aframe-physics-system with a single point-to-point constraint: https://github.com/donmccurdy/aframe-physics-system/blob/master/examples/constraints.html#L74-L91

I'd suggest starting there, even though it will look a little blocky. If you can get that working, there are ways to make it look more like a single continuous rope.

**Comment by gftruj** [comment]
Cannon.js has a spring by itself, you can encapsulate it within an a-frame component - let me know if what i incuded [here](https://stackoverflow.com/questions/51650922/how-to-create-a-spring-in-a-frame-and-cannon-js/51650923#51650923) is satistactory :)

tldr: fiddle [here](https://jsfiddle.net/gftruj/mjo4r1Lc/)

**Comment by Geli25** [comment]
Thanks a lot for your help! I was able to get this working :)

**Comment by gftruj** [comment]
@Geli25  thanks for the feedback, i guess the issue can be closed then :)
btw I'd like to make a legit component of out this on the weekend, my [temporary stuff](https://github.com/gftruj/aframe-physics-spring) should already be sufficient. You can use it if it makes things easier for you.

**Comment by donmccurdy** [comment]
Glad you were able to get it working, and thanks @gftruj! I didn't realize cannon had a spring built in. 😅
